### ISSUE TITLE
FE doesn't send impersonations in API request when changing permissions in a specific way

### ISSUE THREAD

**Issue by noahmoss** [metabase/metabase#38943]
### Describe the bug

Odd edge case in the permissions admin page.

Changing perms for a DB from `Unrestricted` data access + `No` native query editing to `Impersonated` data access and `Yes` native query editing results in the new impersonation policy not being sent in the API request.

https://www.loom.com/share/f21d5e1383b647bf98f52ff6ce335165?sid=1857410b-5b34-4b79-b76c-aad5ebc2052b

### To Reproduce

See loom

### Information about your Metabase installation

```JSON
Current master
```

### Severity

P3

**Comment by paoliniluis** [comment]
@noahmoss but does the FE says it’s impersonated?

**Comment by noahmoss** [comment]
@paoliniluis No, in this situation, the FE reverts it back to "Unrestricted"

**Comment by luizarakaki** [comment]
Unable to reproduce. After the data permissions split, this UI changed. Tried to reproduce using view data + create queries, but couldn't repro
